src/panels/ActionsPanel/CharmActionsPanel/ConfirmationDialog/ConfirmationDialog.test.tsx

```diff
@@ -0,0 +1,201 @@
+      exposed: false,
+      "charm-url": "ch:ceph",
+      "owner-tag": "",
+      life: "alive",
+      "min-units": 0,
+      constraints: { arch: "amd64" },
+      subordinate: false,
+      status: {
+        current: "active",
+        message: "",
+        since: "2023-01-26T06:41:05.303171453Z",
+        version: "",
+      },
+      "workload-version": "",
+    },
+  ];
+  let mockSetConfirmType: Mock;
+  let mockOnRemovePanelQueryParams: Mock;
+
+  beforeAll(() => {
+    state = rootStateFactory.build({
+      juju: jujuStateFactory.build({
+        charms: [
+          charmInfoFactory.build({
+            url: "ch:ceph",
+            actions: {
+              specs: {
+                "add-disk": charmActionSpecFactory.build({
+                  params: applicationCharmActionParamsFactory.build({
+                    properties: {
+                      bucket: {
+                        type: "string",
+                      },
+                      "osd-devices": {
+                        type: "string",
+                      },
+                    },
+                    required: ["osd-devices"],
+                    title: "add-disk",
+                    type: "object",
+                  }),
+                }),
+              },
+            },
+          }),
+        ],
+      }),
+    });
+    mockSetConfirmType = vi.fn();
+    mockOnRemovePanelQueryParams = vi.fn();
+  });
+
+  afterEach(() => {
+    vi.resetModules();
+    vi.restoreAllMocks();
+  });
+
+  it("should return null if there is no selected action", () => {
+    const {
+      result: { container },
+    } = renderComponent(
+      <ConfirmationDialog
+        confirmType={ConfirmType.SUBMIT}
+        selectedAction={undefined}
+        selectedApplications={[]}
+        setConfirmType={vi.fn()}
+        selectedActionOptionValue={undefined}
+        onRemovePanelQueryParams={vi.fn()}
+      />,
+    );
+    expect(container.tagName).toBe("DIV");
+    expect(container.children.length).toBe(1);
+    expect(container.firstChild).toBeEmptyDOMElement();
+  });
+
+  it("should display submit confirmation dialog and can cancel running selected action", async () => {
+    renderComponent(
+      <ConfirmationDialog
+        confirmType={ConfirmType.SUBMIT}
+        selectedAction={"stdout"}
+        selectedApplications={[{ "unit-count": 3 }, { "unit-count": 1 }]}
+        setConfirmType={mockSetConfirmType}
+        selectedActionOptionValue={undefined}
+        onRemovePanelQueryParams={vi.fn()}
+      />,
+    );
+
+    expect(
+      screen.getByRole("dialog", { name: "Run stdout?" }),
+    ).toBeInTheDocument();
+    const cancelButton = screen.getByRole("button", {
+      name: Label.CANCEL_BUTTON,
+    });
+    expect(cancelButton).toBeInTheDocument();
+    expect(
+      screen.getByRole("button", { name: Label.CONFIRM_BUTTON }),
+    ).toBeInTheDocument();
+    expect(
+      screen.getByText("APPLICATION COUNT (UNIT COUNT)"),
+    ).toBeInTheDocument();
```

Comment:
I'm not sure what we gain by testing this. What fail condition are you trying to catch?